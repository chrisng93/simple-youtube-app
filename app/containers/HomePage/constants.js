export const SEARCH_VIDEOS = 'SEARCH_VIDEOS';
export const SEARCH_VIDEOS_SUCCESS = 'SEARCH_VIDEOS_SUCCESS';
export const SEARCH_VIDEOS_FAILURE = 'SEARCH_VIDEOS_FAILURE';

export const SELECT_VIDEO = 'SELECT_VIDEO';
