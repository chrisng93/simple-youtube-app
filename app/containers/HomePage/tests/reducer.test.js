import { fromJS } from 'immutable';
import { SELECT_VIDEO } from '../constants';
import HomeReducer from '../reducer';

describe('HomeReducer', () => {
  it('returns the initial state', () => {
    expect(HomeReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles select video', () => {
    const testAction = {
      type: SELECT_VIDEO,
      payload: { video: { test: 'test' } },
    };
    expect(HomeReducer(undefined, testAction).get('selectedVideo')).toEqual(fromJS(testAction.payload.video));
  });
});
