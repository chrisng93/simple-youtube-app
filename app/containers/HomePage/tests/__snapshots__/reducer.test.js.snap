exports[`HomeReducer returns the initial state 1`] = `
Object {
  "error": Object {
    "message": "",
    "status": false,
  },
  "isSearchingVideos": false,
  "selectedVideo": Object {},
  "videos": Array [],
}
`;
